fastly/resource_fastly_service_v1_splunk_test.go

```diff
@@ -0,0 +1,404 @@
+    priority  = 10
+    type      = "RESPONSE"
+  }
+
+  condition {
+    name      = "ok_response_2XX"
+    statement = "resp.status >= 200 && resp.status < 300"
+    priority  = 10
+    type      = "RESPONSE"
+  }
+
+  splunk {
+    name               = "test-splunk-1"
+    url                = "https://mysplunkendpoint.example.com/services/collector/event"
+    token              = "test-token"
+    format             = "%%h %%l %%u %%{now}V %%{req.method}V %%{req.url}V %%>s %%{resp.http.Content-Length}V"
+    format_version     = 2
+    placement          = "waf_debug"
+    response_condition = "error_response_5XX"
+  }
+
+  splunk {
+    name               = "test-splunk-2"
+    url                = "https://mysplunkendpoint.example.com/services/collector/event"
+    token              = "test-token"
+    format             = "%%h %%l %%u %%{now}V %%{req.method}V %%{req.url}V %%>s %%{resp.http.Content-Length}V"
+    format_version     = 2
+    placement          = "waf_debug"
+    response_condition = "ok_response_2XX"
+  }
+
+  force_destroy = true
+}`, serviceName, domainName)
+}
+
+func testAccServiceV1SplunkConfig_default(serviceName string) string {
+	domainName := fmt.Sprintf("fastly-test.tf-%s.com", acctest.RandString(10))
+
+	return fmt.Sprintf(`
+resource "fastly_service_v1" "foo" {
+  name = "%s"
+
+  domain {
+    name    = "%s"
+    comment = "tf-testing-domain"
+  }
+
+  backend {
+    address = "aws.amazon.com"
+    name    = "tf-test-backend"
+  }
+
+  splunk {
+    name  = "test-splunk"
+    url   = "https://mysplunkendpoint.example.com/services/collector/event"
+    token = "test-token"
+  }
+
+  force_destroy = true
+}`, serviceName, domainName)
+}
+
+func testAccServiceV1SplunkConfig_env(serviceName string) string {
+	domainName := fmt.Sprintf("fastly-test.tf-%s.com", acctest.RandString(10))
+
+	return fmt.Sprintf(`
+resource "fastly_service_v1" "foo" {
+  name = "%s"
+
+  domain {
+    name    = "%s"
+    comment = "tf-testing-domain"
+  }
+
+  backend {
+    address = "aws.amazon.com"
+    name    = "tf-test-backend"
+  }
+
+  splunk {
+    name = "test-splunk"
+    url  = "https://mysplunkendpoint.example.com/services/collector/event"
+  }
+
+  force_destroy = true
+}`, serviceName, domainName)
+}
+
+func setSplunkEnv(token string, t *testing.T) func() {
+	e := getSplunkEnv()
+	// Set all the envs to a dummy value
+	if err := os.Setenv("FASTLY_SPLUNK_TOKEN", token); err != nil {
+		t.Fatalf("Error setting env var FASTLY_SPLUNK_TOKEN: %s", err)
+	}
+
+	return func() {
+		// re-set all the envs we unset above
+		if err := os.Setenv("FASTLY_SPLUNK_TOKEN", e.Token); err != nil {
+			t.Fatalf("Error resetting env var FASTLY_SPLUNK_TOKEN: %s", err)
+		}
+	}
+}
+
+// struct to preserve the current environment
+type currentSplunkEnv struct {
+	Token string
+}
+
+func getSplunkEnv() *currentSplunkEnv {
+	// Grab the existing Fastly Azure SAS token and preserve, in the off chance
```

Comment:
This needs to be changed and not to mentioned Azure, I thought I was experiencing Déjà vu :)